### ISSUE TITLE
[JXR-87] Change line number anchor name pattern

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-jxr#193]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** opened **[JXR-87](https://issues.apache.org/jira/browse/JXR-87?redirect=false)** and commented

Linenumber achor anames are solely the line number but this brings problems. Then linking in APT to source code, it complains:

```
[WARNING] [APT Parser] Modified invalid link: '77' to '../kcc-core/xref/package/Object.html#a77'
```

This means that I am not able to link to lines anymore. Change the anchor name pattern to L\d+

---

**Affects:** 2.2

**Attachments:**
- [JXR-87.patch](https://issues.apache.org/jira/secure/attachment/12716824/JXR-87.patch) (_754 bytes_)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

The suspicious line is: http://svn.apache.org/viewvc/maven/jxr/tags/jxr-2.2/maven-jxr/src/main/java/org/apache/maven/jxr/JavaCodeTransform.java?view=markup#l375

**Comment by jira-importer** [comment]
**[Neeme Praks](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=neeme)** commented

Your patch will break all the links out there that rely on the current format. I would suggest to include a configuration option to restore the old behavior for people that cannot update all the other systems that rely on the old format. But new format should be default as the current format is broken (see also http://stackoverflow.com/questions/7110556/is-it-valid-to-use-only-digits-as-uri-fragment-identifier)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

True, but it was broken anyway. Doxia was broken too before Maven 2.2.x. That's the reason I created this patch. Btw, for those who upgrade from Maven 2.0 to 2.2 they see broken links too. So in my opinion there is little harm.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Is anyone opposed to fix this after more than two years? I will apply this patch and target it for the next release. Those who care for compat should read CHANGES anyway.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Retargeting fix for 2.4.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with r1544832.
